src/modules/search/components/searchbar/index.tsx

```diff
@@ -1,33 +1,91 @@
 'use client';
-import React, { useState, FormEvent } from 'react';
 
-const SearchBar: React.FC = () => {
+import { useDictionary } from '@lib/context/dictionary-context';
+import React, { useState, FormEvent, useEffect, Fragment } from 'react';
+import { ProductPreviewType } from 'types/global';
+import { getProductsList } from '@lib/data/ecommerce';
+import SearchResult from '../search-result';
+import { Transition } from '@headlessui/react';
+import { useDebounce } from '@lib/hooks/use-debounce';
+
+const SearchBar = () => {
+  const dictionary = useDictionary();
+  const [searchDropdownOpen, setSearchDropdownOpen] = useState<boolean>(false);
   const [searchTerm, setSearchTerm] = useState<string>('');
+  const [searchResults, setSearchResults] = useState<ProductPreviewType[]>([]);
+  const debouncedSearch = useDebounce(searchTerm, 400);
```

Comment:
400 is quite a short time for debounce, we should set it to 1000ms to see how it will be in production